### DISCUSSION TITLE
Moes MS-105Z Zigbee Dimmer "Power On State"

### DISCUSSION THREAD

**Discussion by cameronjust** [Koenkk/zigbee2mqtt#12549] [Devices]
Been trying to work out how to set the "on power state" for this device but not having much luck.

https://www.zigbee2mqtt.io/devices/MS-105Z.html

When power is applied to the device it defaults to the light being "off" which is less than ideal. I'm hoping to find a setting somewhere to change it to "previous state" or "on" when power is applied.

My goal is to have it connected to a "normal" rocker switch so it can be used as a normal light switch in case of home automation outages. If left powered I can control in the normal home automation manner.

I'm willing to dig deep if someone wants to give me some tips on how to get into the guts of the settings.

I saw mention of a power on state here but can't find it either in Zigbee2MQTT or Home Assistant - https://smarthomescene.com/reviews/moes-zigbee-2-gang-switch-module-review/

**Comment by cameronjust** [comment]
I was poking around here

https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/moes.js


And found a similar Moe device that does support it (apologies for formatting)

```
    {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_amp6tsvy'}, {modelID: 'TS0601', manufacturerName: '_TZE200_tviaymwx'}],
        model: 'ZTS-EU_1gang',
        vendor: 'Moes',
        description: 'Wall touch light switch (1 gang)',
        exposes: [e.switch().setAccess('state', ea.STATE_SET),
            exposes.enum('indicate_light', ea.STATE_SET, Object.values(tuya.moesSwitch.indicateLight))
                .withDescription('Indicator light status'),
            exposes.enum('power_on_behavior', ea.STATE_SET, Object.values(tuya.moesSwitch.powerOnBehavior))
                .withDescription('Controls the behavior when the device is powered on')],
        fromZigbee: [fz.tuya_switch, fz.moes_switch],
        toZigbee: [tz.tuya_switch_state, tz.moes_switch],
        onEvent: tuya.onEventSetLocalTime,
        configure: async (device, coordinatorEndpoint, logger) => {
            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
            // Reports itself as battery which is not correct: https://github.com/Koenkk/zigbee2mqtt/issues/6190
            device.powerSource = 'Mains (single phase)';
            device.save();
        },
    },
```

So the original device I'm using's code is this

```
    {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_la2c2uo9'}],
        model: 'MS-105Z',
        vendor: 'Moes',
        description: '1 gang 2 way Zigbee dimmer switch',
        fromZigbee: [fz.moes_105_dimmer, fz.ignore_basic_report],
        toZigbee: [tz.moes_105_dimmer],
        meta: {turnsOffAtBrightness1: true},
        configure: async (device, coordinatorEndpoint, logger) => {
            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff', 'genLevelCtrl']);
        },
        exposes: [e.light_brightness().setAccess('state', ea.STATE_SET).setAccess('brightness', ea.STATE_SET)],
    },
```

would it be as simple as modifying it to this?

```
    {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_la2c2uo9'}],
        model: 'MS-105Z',
        vendor: 'Moes',
        description: '1 gang 2 way Zigbee dimmer switch',
        fromZigbee: [fz.moes_105_dimmer, fz.ignore_basic_report, tz.moes_switch],
        toZigbee: [tz.moes_105_dimmer, tz.moes_switch],
        meta: {turnsOffAtBrightness1: true},
        configure: async (device, coordinatorEndpoint, logger) => {
            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff', 'genLevelCtrl']);
        },
        exposes: [e.light_brightness().setAccess('state', ea.STATE_SET).setAccess('brightness', ea.STATE_SET),
            exposes.enum('power_on_behavior', ea.STATE_SET, Object.values(tuya.moesSwitch.powerOnBehavior))
                .withDescription('Controls the behavior when the device is powered on')]],
    },
```

Only problem I have now is I'm running zigbee2mqtt in a docker container so not sure where this file would be to edit or even if I can edit it.

Edit: Found it in /app/node_modules/zigbee-herdsman-converters/devices/moes.js of the docker container.

Modified and restarted the container and it didn't work. No new settings in zigbeemqtt.

Light still works though so I didn't break anything which is a bonus.

**Comment by juagicre** [comment]
Nothing new here? I guess having this product on the "supported devices" is quite right as it is detected, paired and should be working, but a big warning should be somewhere when it depends on also other devices to be connected to it... so, many people buying rele dimmers are falling into the trap of having to buy as well the dimmer switch, which to my understanding is quite avoidable if you have he rele device and you just want to play with the dimmer under some circumstances and rest of the time use it as "ON/OFF" so, as a normal switch rele. 
Maybe here there is a hardware limitation and that cannot be done? If somebody knows of an alternative that works with normal switches and is being sold at a decent price on aliexpress, please, post here names, links and ideas :)
